com.unity.ml-agents/Runtime/Sensors/RenderTextureSensorComponent.cs

```diff
@@ -78,8 +97,15 @@ public override int[] GetObservationShape()
         {
             var width = RenderTexture != null ? RenderTexture.width : 0;
             var height = RenderTexture != null ? RenderTexture.height : 0;
+            var observationShape = new[] { height, width, Grayscale ? 1 : 3 };
+
+            var stacks = ObservationStacks > 1 ? ObservationStacks : 1;
+            if (stacks > 1)
+            {
+                observationShape[observationShape.Length - 1] *= stacks;
```

Comment:
nit: you can use `observationShape[2]` - no need to generalize since you make observationShape above.